### PR TITLE
Add metric counters to track sensor data in the AddSensorDataBatchHandler class

### PR DESCRIPTION
**Pull request by jkammerl**
Adds additional metrics to track incoming sensor and local slam result data in Prometheus.

### CODE REVIEW

File: cartographer/metrics/family_factory.h [hunk 1]
```diff
@@ -27,14 +27,31 @@
...
 namespace metrics {
 
+template <typename MetricType>
+class NullFamily;
+
 template <typename MetricType>
 class Family {
- public:
+ public:  // Family instance that does nothing. Safe for use in static
+          // initializers.
```
Comment (pifon2a) [inline cartographer/metrics/family_factory.h]: nit: new line after `public:`

**Comment by gaschler** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user gaschler. Merge queue now has a length of 1.

**Comment by MichaelGrupp** [comment]
@jkammerl Is it normal that this logs so frequently?

```
I0803 12:49:17.917871 44537 add_sensor_data_batch_handler.cc:128] Create metrics handler for client: toru-0013
I0803 12:49:18.681766 44537 add_sensor_data_batch_handler.cc:128] Create metrics handler for client: toru-0013
I0803 12:49:19.435117 44537 add_sensor_data_batch_handler.cc:128] Create metrics handler for client: toru-0013
I0803 12:49:20.180188 44539 add_sensor_data_batch_handler.cc:128] Create metrics handler for client: toru-0013
I0803 12:49:20.980387 44539 add_sensor_data_batch_handler.cc:128] Create metrics handler for client: toru-0013
I0803 12:49:21.737989 44540 add_sensor_data_batch_handler.cc:128] Create metrics handler for client: toru-0013
```

**Comment by cschuet** [comment]
@MichaelGrupp Agreed. This is too much output. I will remove the `LOG(INFO)`.

**Comment by cschuet** [comment]
@MichaelGrupp https://github.com/googlecartographer/cartographer/pull/1366

**Comment by MichaelGrupp** [comment]
Thanks. I was not sure if it's normal that this "metrics handler" is added so often.

**Pull request merged**